.github/workflows/generate-pinval.yaml

```diff
@@ -149,57 +184,34 @@ jobs:
       - name: Install Hugo
         run: sudo dpkg -i ${{ runner.temp }}/hugo.deb
 
-      - name: Generate PINVAL reports (triad shard or explicit PINs)
-        run: |
-          set -euo pipefail
-
-          declare -a ARGS=(
-            --run-id "${{ inputs.run_id }}"
-            --triad "${{ inputs.triad_name }}"
-            --township "$TOWNSHIP"
-          )
-
-          # Convert comma-separated input into an array, stripping blanks
-          IFS=',' read -ra PINS_ARR <<< "${{ inputs.pins }}"
-          CLEAN_PINS=()
-          for p in "${PINS_ARR[@]:-}"; do
-            [[ -n "$p" ]] && CLEAN_PINS+=("$p")
-          done
-
-          if [[ ${#CLEAN_PINS[@]} -gt 0 ]]; then
-            ARGS+=(--pin "${CLEAN_PINS[@]}")
-          fi
-
-          echo "Running generate_pinval.py with: ${ARGS[*]}"
-          python3 scripts/generate_pinval/generate_pinval.py "${ARGS[@]}"
-
-      - name: Resolve target S3 bucket
-        id: bucket
+      - name: Generate PINVAL reports
+        env:
+          RUN_ID: ${{ needs.setup.outputs.run-id }}
         run: |
-          if [[ "${{ inputs.environment }}" == 'Development bucket' ]]; then
-            echo "bucket=${{ vars.AWS_S3_BUCKET_PINVAL_STAGING }}" >> "$GITHUB_OUTPUT"
-          else
-            echo "bucket=${{ vars.AWS_S3_BUCKET_PINVAL_PROD }}" >> "$GITHUB_OUTPUT"
-          fi
+          read -ra PINS <<< "${{ inputs.pins }}"
+          python3 scripts/generate_pinval/generate_pinval.py \
+            --run-id "$RUN_ID" \
+            --township "$TOWNSHIP" \
+            --pin "${PINS[@]}"
 
       - name: Sync reports to S3
         env:
-          TARGET_BUCKET: ${{ steps.bucket.outputs.bucket }}
+          TARGET_BUCKET: ${{ needs.setup.outputs.bucket }}
+          YEAR: ${{ needs.setup.outputs.assessment-year }}
         run: |
-          RUN_ID="${{ inputs.run_id }}"
-          YEAR="${RUN_ID:0:4}"
           LOCAL="hugo/public/${YEAR}/"
           REMOTE="${TARGET_BUCKET%/}/${YEAR}/"
           aws s3 sync "$LOCAL" "$REMOTE" \
             --no-progress \
             --include '*.html'
 
 ###############################################################################
-# SNS Job
+# Invalidate cache and notify
 ###############################################################################
   notify:
-    needs: generate-pinval
-    if: ${{ always() && needs.generate-pinval.result == 'success' }}
```

Comment:
By default, workflow jobs will only execute if their dependencies succeed, so I don't actually think we need this conditional.